### PR TITLE
log: Write more details in write_pidfile

### PR DESCRIPTION
**Pull request by Snorch**
This is a backport of:
https://src.openvz.org/projects/OVZ/repos/criu/commits/8e6bc1aeb

Signed-off-by: Cyrill Gorcunov <gorcunov@virtuozzo.com>
Signed-off-by: Pavel Tikhomirov <ptikhomirov@virtuozzo.com>

### CODE REVIEW

File: criu/log.c [hunk 1]
```diff
@@ -397,15 +397,22 @@ void print_on_level(unsigned int loglevel, const char *format, ...)
...
 
-	dprintf(fd, "%d", pid);
+	ret = dprintf(fd, "%d", pid);
+	if (ret < 0) {
+		pr_perror("pidfile: Can't write pid %d to %s", pid, opts.pidfile);
+		close(fd);
+		return -1;
+	}
+
+	pr_debug("pidfile: Wrote pid %d to %s (%d bytes)\n", pid, opts.pidfile, ret);
```
Comment (adrianreber) [inline criu/log.c:415]: Looking at the return at the end of the function which returns -1 in the case of ret being 0 it is probably a bit strange to have this `pr_debug()` line if 0 bytes were written.

**Comment by avagin** [comment]
pls remove a link to external resources, it doesn't add any value here, you can add it in a comment to this pr.

File: criu/log.c (hunk 1 above)
Comment (Snorch) [reply criu/log.c:419]: Sure, making all cases separate has more logic here as we need to print differently in different cases.

**Comment by Snorch** [comment]
> pls remove a link to external resources, it doesn't add any value here, you can add it in a comment to this pr.

Ok. I thought about it not as a valuable info, but more of giving a credit to the original source.

**Pull request merged**